keras/src/distillation/distiller_test.py

```diff
@@ -0,0 +1,526 @@
+
+        # Check predictions sum to reasonable values (not zeros/infinities)
+        prediction_sums = np.sum(predictions, axis=1)
+        self.assertTrue(np.all(np.isfinite(prediction_sums)))
+
+    def test_distiller_serialization_and_saving(self):
+        """Test Distiller serialization, saving, and loading."""
+
+        # Use standard Sequential models for serialization testing
+        teacher = keras.Sequential(
+            [
+                keras.layers.Dense(
+                    32, activation="relu", name="teacher_dense_1"
+                ),
+                keras.layers.Dense(
+                    16, activation="relu", name="teacher_dense_2"
+                ),
+                keras.layers.Dense(10, name="teacher_output"),
+            ]
+        )
+
+        student = keras.Sequential(
+            [
+                keras.layers.Dense(
+                    16, activation="relu", name="student_dense_1"
+                ),
+                keras.layers.Dense(
+                    8, activation="relu", name="student_dense_2"
+                ),
+                keras.layers.Dense(10, name="student_output"),
+            ]
+        )
+
+        # Create distiller with single strategy
+        strategy = LogitsDistillation(temperature=3.0, loss="kl_divergence")
+
+        original_distiller = Distiller(
+            teacher=teacher,
+            student=student,
+            strategies=strategy,
+            student_loss_weight=0.7,
+        )
+
+        # Build the models by calling them
+        x_test = np.random.random((2, 20)).astype(np.float32)
+        _ = original_distiller(x_test)
+
+        # Test get_config
+        config = original_distiller.get_config()
+
+        # Verify all components are in config
+        required_keys = [
+            "teacher",
+            "student",
+            "strategies",
+            "strategy_weights",
+            "student_loss_weight",
+        ]
+        for key in required_keys:
+            self.assertIn(key, config, f"Missing key: {key}")
+
+        # Test JSON serialization
+        json_str = json.dumps(config)
+        self.assertIsInstance(json_str, str)
+
+        # Test from_config reconstruction
+        reconstructed_distiller = Distiller.from_config(config)
+
+        # Verify reconstruction
+        self.assertEqual(reconstructed_distiller.student_loss_weight, 0.7)
+        self.assertIsInstance(
+            reconstructed_distiller.strategies[0], LogitsDistillation
+        )
+
+        # Verify strategy parameters
+        self.assertEqual(reconstructed_distiller.strategies[0].temperature, 3.0)
+
+        # Test that reconstructed distiller can be used for inference
+        reconstructed_output = reconstructed_distiller(x_test)
+        self.assertEqual(reconstructed_output.shape, (2, 10))
+
+        # Test model saving and loading (full integration test)
+        with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
nitpick: You don't need a `with` and indented code if you just use `self.get_temp_dir()`, it cleans up automaticatlly.